Bluejay/Bluejay/Bluejay.swift

```diff
@@ -1008,7 +1008,7 @@ extension Bluejay: CBCentralManagerDelegate {
         }
         
         for observer in observers {
-            observer.weakReference?.disconnected(from: disconnectedPeripheral)
+            observer.weakReference?.disconnected(from: disconnectedPeripheral, with: error)
```

Comment:
Now that we include the error, there is an even higher chance that people will be tempted and led to writing some kind reconnection logic in their `disconnected` event callbacks. That wouldn't work as it is right now, because at this point we haven't reached the `disconnectCleanUp` yet, so any new `Connection` operation being added to the queue during the `disconnect` event will fail immediately as Bluejay will still think we're connected.

We may have to move this entire for-loop block after line 1053, after niling out `connectingPeripheral` and `connectedPeripheral`. But there are several problems with that as well:
1. Unclear if moving the notification of observers will have any negative impact, but it certainly does change the whole procedures here as the order of operations is really important. We will have to be very careful as it could break some apps in production.
- One thing that will happen for sure if we move the for-loop clause is that you will get a `notConnected` error in your in-flight and already queued operations, such as any recently called reads or writes, and then your register observers will get the `disconnected` event last. That's not necessarily wrong, but it's a difference.
2. If we are expecting users to write their own reconnection logic, then we need to force them to handle Bluejay's auto reconnection at this step as well - so in the disconnection event, the user needs to be able to:
  - Check whether Bluejay will automatically reconnect, if so, maybe no need to write your own reconnection
  - Ability to cancel Bluejay's automatic reconnect, so users can either use their own reconnection code or just ask Bluejay to not try to reconnect
  - Maybe all of this means a 3rd parameter for the `disconnected` delegation of a callback type with a default value (so it's optional) that will then be invoked inside `disconnectCleanUp`. Not sure yet, will have to think a bit more.

Reply:
Sorry, for the first file totally useless.
About your considerations, yep it's all true. I also had this issue after disconnecting and trying to reconnect on another device same issue as the user reported in [#168](https://github.com/steamclock/bluejay/issues/168). 
I will look into that and try to propose a solution.